docs/development/machine-bootstrap.md

```diff
@@ -0,0 +1,19 @@
+# Machine-Controller integration into Gardener
+The Gardener is deploying the [machine-controller-manager](https://github.com/gardener/machine-controller-manager) as part of a Shoot controlplane onto a Seed to manage the nodes/machines of the Shoot cluster. The machine-controller-manager will create, bootstrap, update and delete the nodes of the Shoot cluster and the underlying virtual machines.
+
+## Flow of creating and bootstrapping Machines/Nodes
+The gardener-controller-manager will ensure during the creation/reconciliation flow
+- that the machine-controller-manager gets deployed into the Shoot namespace of the Seed.
+- that proper ``<cloudprovider>machineclass`` objects and its corresponding secrets exists in the Shoot namespace of the Seed. Those secrets contain information to bootstrap the node and will be copied to the machines of the related class.
+- that ``machine-deployments`` for the in the Shoot manifest specified worker groups exists. Those deployments will spawn the required ``machine`` objects in the Shoot namespace of the Seed.
+- that a valid bootstrap-token secret exists in the kube-system namespace of the Shoot.
+- that the cloud-config configuration for setting up nodes exists in the kube-system namespace of the Shoot.
+
+The machine-controller-manager will pick up the ``machine`` objects and create the vms on the respective infrastructure provider. The bootstrap of a vm, which should join the Shoot cluster as node, is done via CoreOS [cloud-config](https://coreos.com/os/docs/latest/cloud-config.html).
```

Comment:
s/``machine``/`Machine`

Reply:
s/vms/VMs

Reply:
s/infrastructure/cloud

Reply:
s/The bootstrap of a VM.../Bootstrapping VMs is done via OS-specific configuration (e.g., [CoreOS cloud-config](https://coreos.com/os/docs/latest/cloud-config.html)).